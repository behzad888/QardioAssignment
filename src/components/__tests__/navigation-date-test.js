//@flow
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import NavigationDate from '../navigation-date/navigation-date';

describe('while navigation date test running in a browser environment', () => {
  let container, wrapper;

  beforeEach(() => {
    jest.resetModules();

    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    ReactDOM.render(null, container);
    document.body.removeChild(container);
    container = null;
  });

  it('should render correctly', () => {
    wrapper = renderer
      .create(
        <div>
        <NavigationDate />
        <NavigationDate  className="navigation-date-test" />
        <NavigationDate   className="navigation-date-test" title={new Date().toDateString()}/>
        <NavigationDate   isBusy={true}/>
        </div>,
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});