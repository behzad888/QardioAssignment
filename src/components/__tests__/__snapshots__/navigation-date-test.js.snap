// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`while navigation date test running in a browser environment should render correctly 1`] = `
<div>
  <div
    className="navigation-date "
  >
    <span
      className="title"
    >
      Mon Jul 01 2019
    </span>
    <button
      className="btn small btn-White ghost round"
      onClick={[Function]}
    >
      &lt;
    </button>
    <button
      className="btn small btn-White ghost round"
      onClick={[Function]}
    >
      &gt;
    </button>
  </div>
  <div
    className="navigation-date navigation-date-test"
  >
    <span
      className="title"
    >
      Mon Jul 01 2019
    </span>
    <button
      className="btn small btn-White ghost round"
      onClick={[Function]}
    >
      &lt;
    </button>
    <button
      className="btn small btn-White ghost round"
      onClick={[Function]}
    >
      &gt;
    </button>
  </div>
  <div
    className="navigation-date navigation-date-test"
  >
    <span
      className="title"
    >
      Mon Jul 01 2019
    </span>
    <button
      className="btn small btn-White ghost round"
      onClick={[Function]}
    >
      &lt;
    </button>
    <button
      className="btn small btn-White ghost round"
      onClick={[Function]}
    >
      &gt;
    </button>
  </div>
  <div
    className="navigation-date "
  >
    <span
      className="title"
    >
      Mon Jul 01 2019
    </span>
    <button
      className="btn small btn-White ghost round"
      onClick={[Function]}
    >
      ...
    </button>
    <button
      className="btn small btn-White ghost round"
      onClick={[Function]}
    >
      ...
    </button>
  </div>
</div>
`;
